### DISCUSSION TITLE
Newbie here: how to view the post-babel code that is generated by twin.macro?

### DISCUSSION THREAD

**Discussion by ChristineTham** [ben-rogerson/twin.macro#278] [Q&A] [answered]
Newbie question:
Is it possible to view the transformed version (for styled-components / emotion) of the source file? I gather I can invoke babel-cli with the right options and view the output file, but being a newbie I don't know how to do this. How were the output examples in the documentation generated?

Another newbie question:
I am thinking it may be cool to transform a Tailwind component into theme-ui or chakra-ui and I was hoping there can be an automated way of doing that. Is this possible, or am I better off just writing the component from scratch using the Tailwind as a reference rather than automatic translation?

**Comment by ben-rogerson** [comment, marked as answer]
Great questions!

To view the transformed output you need to run the file through babel.

Once you’ve setup twin, you can run babel over the file like this:

```shell
npx babel index.js --out-file index-compiled.js
```

But to make this easier, you can use the [twin-output-tester](https://github.com/ben-rogerson/twin-output-tester) which is all setup and ready to go.

> I am thinking it may be cool to transform a Tailwind component into theme-ui or chakra-ui and I was hoping there can be an automated way of doing that. Is this possible, or am I better off just writing the component from scratch using the Tailwind as a reference rather than automatic translation?

I believe you’ll need to convert the styling in those components manually as the code varies too much between those UI frameworks. 
If you use the tester above, you'll at least be able to copy and paste some of the code.

**Reply by ChristineTham** [reply]
Thank you so much. And thanks for creating an awesome utility!
